packages/apps/job-launcher/server/src/modules/job/job.service.ts

```diff
@@ -101,28 +111,312 @@ export class JobService {
+        requester_question: { en: jobDto.annotations.labelingPrompt },
+        job_total_tasks: objectsInBucket.length,
+        task_bid_price: jobDto.annotations.taskBidPrice,
+        taskdata_uri: await this.generateAndUploadTaskData(jobDto.dataUrl, objectsInBucket),
+        public_results: true,
+        oracle_stake: 0.05,
+        repo_uri: "http://reporacle.yellow.svc.cluster.local:5000",
+        ro_uri: "http://recoracle:5000/rocheck"
+    };
+
+    let groundTruthsData;
+    if (jobDto.annotations.groundTruths) {
+      groundTruthsData = await this.storageService.download(jobDto.annotations.groundTruths)
+
+      if (isValidJSON(groundTruthsData)) {
+        groundTruthsData = JSON.parse(groundTruthsData);
+      }
+    }
+
+    switch (jobType) {
+        case JobCaptchaShapeType.COMPARISON:
+          return {
+              ...commonManifestProperties,
+              request_type: JobCaptchaRequestType.IMAGE_LABEL_BINARY,
+              groundtruth_uri: jobDto.annotations.groundTruths,
+              restricted_audience: {
+                sitekey: [
+                  {
+                    [this.configService.get<number>(ConfigNames.HCAPTCHA_SITE_KEY)!]: {
+                      score: 1
+                    }
+                  }
+                ]
+              },// this.buildHCaptchaRestrictedAudience(jobDto.advanced),
+              requester_restricted_answer_set: {},
+              requester_question_example: jobDto.annotations.exampleImages || [],
+          };
+
+        case JobCaptchaShapeType.CATEGORAZATION:
+          return {
+              ...commonManifestProperties,
+              request_type: JobCaptchaRequestType.IMAGE_LABEL_MULTIPLE_CHOICE,
+              groundtruth_uri: jobDto.annotations.groundTruths,
+              restricted_audience: {
+                sitekey: [
+                  {
+                    [this.configService.get<number>(ConfigNames.HCAPTCHA_SITE_KEY)!]: {
+                      score: 1
+                    }
+                  }
+                ]
+              },// this.buildHCaptchaRestrictedAudience(jobDto.advanced),
+              requester_restricted_answer_set: this.buildHCaptchaRestrictedAnswerSet(groundTruthsData)
+          };
+
+        case JobCaptchaShapeType.POLYGON:
+          if (!jobDto.annotations.label) {
+            this.logger.log(ErrorJob.JobParamsValidationFailed, JobService.name);
+            throw new BadRequestException(ErrorJob.JobParamsValidationFailed);
+          }
+          
+          const polygonManifest = {
+              ...commonManifestProperties,
+              request_type: JobCaptchaRequestType.IMAGE_LABEL_AREA_SELECT,
+              request_config: {
+                  shape_type: JobCaptchaShapeType.POLYGON,
+                  min_shapes_per_image: HCAPTCHA_MIN_SHAPES_PER_IMAGE,
+                  max_shapes_per_image: HCAPTCHA_MAX_SHAPES_PER_IMAGE,
+                  min_points: 4,
+                  max_points: 4,
```

Comment:
why are those values constant?

Reply:
These are the most preferred object settings for hCaptcha. They have been removed to keep the interface simple. Also, their implementation required additional time, since all these values must be correctly validated depending on the selected type. For example Polygon has 4 and 4, Landmark 1 and 8, etc.

Reply:
we should maybe use constants here then